src/RTC_RV3032C7.cpp

```diff
@@ -0,0 +1,317 @@
+#define RV3032C7_CONTROL3 0x12  ///< Control register
+#define RV3032C7_INT_MASK 0x14  ///< Clock Interrupt Mask Register
+#define RV3032C7_TEMPERATUREREG                                                \
+  0x0E ///< Temperature register (bit 4-7 = lowest 4 bits.
+       ///<  high byte is at RV3032C7_TEMPERATURE8BIT), 12-bit
+       ///<  Temperature value
+#define RV3032C7_TEMPERATURE8BIT 0x0F ///< 8-bit temperature value
+
+// RAM Addresses that mirror EEPROM registers
+#define RV3032C7_PMU 0xC0 ///< Power Management Unit (PMU)
+
+// Status register flags
+#define RV3032C7_THF 0x80  ///< Temp. High
+#define RV3032C7_TLF 0x40  ///< Temp. Low
+#define RV3032C7_UF 0x20   ///< Periodic Time Update
+#define RV3032C7_TF 0x10   ///< Periodic Coundown Timer
+#define RV3032C7_AF 0x08   ///< Alarm
+#define RV3032C7_EVF 0x04  ///< External Event
+#define RV3032C7_PORF 0x02 ///< Power On Reset
+#define RV3032C7_VLF 0x01  ///< Voltage Low
+
+// Control register flags (some)
+#define RV3032C7_XBIT 0x20 ///< Control1, "X" bit (must be set to 1)
+#define RV3032C7_EERD 0x04 ///< Control1, ROM Memory Refresh Disable bit.
+#define RV3032C7_STOP 0x01 ///< Control2, STOP bit
+#define RV3032C7_EIE 0x04  ///< Control2, External Event Interrupt Enable bit
+#define RV3032C7_AIE 0x08  ///< Control2, Alarm Interrupt Enable bit
+#define RV3032C7_TIE                                                           \
+  0x10 ///< Control2, Periodic Countdown Timer Interrupt Enable bit
+#define RV3032C7_UIE                                                           \
+  0x20 ///< Control2, Periodic Time Update Interrupt Enable bit
+#define RV3032C7_CLKIE                                                         \
+  0x40 ///< Control2, Interrupt Controlled Clock Output Enable bit
+
+// Clock Interrupt Mask register flags (only those used in this file)
+#define RV3032C7_CAIE 0x10 ///< Clock output when Alarm Interrupt Enable bit
+
+// Power Management Unit (PMU) register flags (only those used in this file)
+#define RV3032C7_NCLKE                                                         \
+  0x40 ///< Not CLKOUT Enable Bit in Power Management Unit (PMU)
+#define RV3032C7_BSM                                                           \
+  0x30 ///< Backup Switchover Mode                                                         \
+
+
+// Temperature register flags (some flags ended up here, albeit unrelated to
+// temperature)
+#define RV3032C7_CLKF 0x02 ///< Clock Output Interrupt Flag (CLKF)
+
+/**************************************************************************/
+/*!
+    @brief  Start I2C for the RV3032C7, test succesful connection
+            and initialize the chip for use with RTClib
+    @param  wireInstance pointer to the I2C bus
+    @return True if Wire can find RV3032C7 or false otherwise.
+*/
+/**************************************************************************/
+boolean RTC_RV3032C7::begin(TwoWire *wireInstance) {
```

Comment:
The [`boolean` type is being phased out][1] by the Arduino team in favor of `bool`: “[It’s recommended to instead use the standard type `bool`, which is identical.][2]”

```suggestion
bool RTC_RV3032C7::begin(TwoWire *wireInstance) {
```

Admittedly, the existing code is inconsistent, as it uses of both `boolean` and `bool`. For new code, thought, it is probably better to stick with the preferred type.

[1]: https://github.com/arduino/Arduino/issues/4673
[2]: https://www.arduino.cc/reference/en/language/variables/data-types/boolean/

Reply:
I just submitted a pull request (#259) for replacing the existing occurrences of `boolean` by `bool`.